examples/carousel/js/carousel-tablist.js

```diff
@@ -0,0 +1,373 @@
+  width3 = this.pauseButtonNode.getBoundingClientRect().width;
+  this.tablistNode.style.left = (((width2-width1)/2) - (3*width3/2)) +'px';
+}
+
+
+CarouselTablist.prototype.getTabpanelNode = function (tabNode) {
+  var index = this.tabNodes.indexOf(tabNode);
+  return this.tabpanelNodes[index];
+}
+
+CarouselTablist.prototype.hideTabpanel = function (tabNode) {
+  tabNode.setAttribute('aria-selected', 'false');
+  tabNode.setAttribute('tabindex', '-1');
+
+  var tabpanelNode = this.getTabpanelNode(tabNode);
+
+  if (tabpanelNode) {
+    tabpanelNode.classList.remove('active');
+  }
+}
+
+CarouselTablist.prototype.showTabpanel = function (tabNode, moveFocus) {
+  if (typeof moveFocus !== 'boolean') {
+    moveFocus = false;
+  }
+
+  tabNode.setAttribute('aria-selected', 'true');
+  tabNode.removeAttribute('tabindex');
+
+  var tabpanelNode = this.getTabpanelNode(tabNode);
+
+  if (tabpanelNode) {
+    tabpanelNode.classList.add('active');
+  }
+
+  if (moveFocus) {
+    tabNode.focus();
+  }
+}
+
+CarouselTablist.prototype.setSelectedTab = function (newTab, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  for (var i = 0; i < this.tabNodes.length; i++) {
+    this.hideTabpanel(this.tabNodes[i]);
+  }
+
+  this.currentTabNode = newTab;
+  this.showTabpanel(newTab, moveFocus);
+}
+
+CarouselTablist.prototype.setSelectedToPreviousTab = function (currentTabNode, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.firstTabNode) {
+    this.setSelectedTab(this.lastTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index - 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedToNextTab = function (currentTabNode, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.lastTabNode) {
+    this.setSelectedTab(this.firstTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index + 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedTofirstTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.firstTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.setSelectedTolastTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.lastTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.rotateSlides = function () {
+  if (this.rotate) {
+    this.setSelectedToNextTab();
+  }
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
```

Comment:
this is just a minor note -- this particular implementation (over a `setTimeout`/`clearTimeout` one) works, although it means this function will continue to be called even when paused.

Reply:
@smhigley 

Since `this.rotate` is controlled by a number of events and situtation, to me it was just seemed the best way to control the rotation.   For example rotation is paused and started based on hover and keyboard focus, do I need to keep track of all the different ways and conditions to restart rotation?  This seemed easier for me to understand.